nova/virt/vmwareapi/vmops.py

```diff
@@ -4332,8 +4332,23 @@ def _sync_sync_server_group(context, sg_uuid):
             LOG.debug('Sync for server-group %s done', sg_uuid)
 
         def _update_rule(rule_name, expected_members, sg):
-            rule = cluster_util.get_rule(
-                self._session, self._cluster, rule_name)
+            # we need to get by "prefix", to get all rules matching our name,
+            # as there can be duplication happening with automatically
+            # vSphere-created rules during vMotion
+            rules = [r for r in cluster_util.get_rules_by_prefix(
+                        self._session, self._cluster, rule_name)
+                     if r.name == rule_name]
+
+            rule = rules[0] if rules else None
+
+            # if we have duplicates (with the same name), delete them
+            for dupl_rule in rules[1:]:
```

Comment:
How about keeping the rule with the most VMs?

```suggestion
            for dupl_rule in sorted(rules, key=lambda r: len(r.vm), reverse=True)[1:]:
```

I know it probably has little bearing on how much work the sync-loop has to do. I'm just being cautious with regards to what behind-the-scenes work an "add a vm to a rule" might incur in VMware.

Reply:
I'd rather keep the Nova-created rule instead of some VMware-created one. My assumption is, that the rule with the lowest key is the nova-created one. This should also be the one Nova keeps up-to-date and thus should™ be the one with the most VMs.